src/dune_rules/coq/coq_rules.ml

```diff
@@ -572,7 +572,16 @@ let setup_theory_rules ~sctx ~dir ~dir_contents (s : Coq_stanza.Theory.t) =
   let theory_dirs =
     Coq_sources.directories coq_dir_contents ~name |> Path.Build.Set.of_list
   in
-  let coq_modules = Coq_sources.library coq_dir_contents ~name in
+  let coq_modules =
+    (* filter out all the load only modules form the module sources list *)
+    Coq_sources.library coq_dir_contents ~name
+    (* We don't generate rules for modules that are only loaded *)
+    |> List.filter ~f:(fun coq_module ->
+           not
+           @@ List.mem ~equal:Coq_module.equal
+                (Coq_sources.load_only_modules coq_dir_contents ~dir s)
+                coq_module)
+  in
```

Comment:
So instead of this let's pass `load_source_rule` and we do for that:
```
    >>> Action_builder.(with_no_targets extra_source_rule)
```
in setup_coqdep_for_theory_rule